crates/sui-framework/packages/deepbook/sources/math.move

```diff
@@ -6,25 +6,104 @@ module deepbook::math {
     const FLOAT_SCALING: u64 = 1_000_000_000;
     const FLOAT_SCALING_U128: u128 = 1_000_000_000;
 
-    public fun mul(x: u64, y: u64): u64 {
-        let x = (x as u128);
-        let y = (y as u128);
-        ((x * y / FLOAT_SCALING_U128) as u64)
+    // <<<<<<<<<<<<<<<<<<<<<<<< Error codes <<<<<<<<<<<<<<<<<<<<<<<<
+    const EUnderflow: u64 = 1;
+    // <<<<<<<<<<<<<<<<<<<<<<<< Error codes <<<<<<<<<<<<<<<<<<<<<<<<
+
+    // multiply two floating numbers
+    public fun unsafe_mul(x: u64, y: u64): u64 {
+        let (_, result) = unsafe_mul_round(x, y);
+        result
     }
 
-    public fun mul_round(x: u64, y: u64): (bool, u64) {
+    // multiply two floating numbers and assert the result is non zero
+    public fun safe_mul(x: u64, y: u64): u64 {
+        let (_, result) = unsafe_mul_round(x, y);
+        assert!(result > 0, EUnderflow);
+        result
+    }
+
+    // multiply two floating numbers
+    // also returns whether the result is rounded down
+    public fun unsafe_mul_round(x: u64, y: u64): (bool, u64) {
         let x = (x as u128);
         let y = (y as u128);
         let is_round_down = true;
         if ((x * y) % FLOAT_SCALING_U128 == 0) is_round_down = false;
         (is_round_down, ((x * y / FLOAT_SCALING_U128) as u64))
     }
 
-    public fun div_round(x: u64, y: u64): (bool, u64) {
+    // multiply two floating numbers and assert the result is non zero
+    // also returns whether the result is rounded down
+    public fun safe_mul_round(x: u64, y: u64): (bool, u64) {
+        let (is_round_down, result) = unsafe_mul_round(x, y);
+        assert!(result > 0, EUnderflow);
+        (is_round_down, result)
+    }
+
+    // divide two floating numbers
+    // also returns whether the result is rounded down
+    public fun unsafe_div_round(x: u64, y: u64): (bool, u64) {
         let x = (x as u128);
         let y = (y as u128);
         let is_round_down = true;
         if ((x * (FLOAT_SCALING as u128) % y) == 0) is_round_down = false;
         (is_round_down, ((x * (FLOAT_SCALING as u128) / y) as u64))
     }
+
+    // divide two floating numbers and assert the result is non zero
+    // also returns whether the result is rounded down
+    public fun safe_div_round(x: u64, y: u64): (bool, u64) {
+        let (is_round_down, result) = unsafe_div_round(x, y);
+        assert!(result > 0, EUnderflow);
+        (is_round_down, result)
+    }
+
+    #[test]
+    fun test_mul() {
+        assert!(unsafe_mul(1_000_000_000, 1) == 1, 0);
```

Comment:
Import `sui::test::assert_eq` and then use

```suggestion
        assert_eq!(unsafe_mul(1_000_000_000, 1), 1);
```

here and elsewhere for better test output

Reply:
Get it!